examples/tutorials/testing.md

```diff
@@ -0,0 +1,729 @@
+---
+title: "Writing tests in Deno: A complete guide"
+description: "Learn key concepts like test setup and structure, assertions, async testing, mocking, test fixtures, parallelization, and code coverage"
+url: /examples/testing_tutorial/
+---
+
+Testing is critical in software development to ensure your code works as
+expected, and continues to work as you make changes. Tests verify that your
+functions, modules, and applications behave correctly, handle edge cases
+appropriately, and maintain expected performance characteristics.
+
+## Table of Contents
+
+- [Why testing matters](#why-testing-matters)
+- [Writing tests with `Deno.test`](#writing-tests-with-denotest)
+- [Test structure and organization](#test-structure-and-organization)
+- [Assertions](#assertions)
+- [Async testing](#async-testing)
+- [Mocking in tests](#mocking-in-tests)
+- [Test parallelization](#test-parallelization)
+- [Coverage](#coverage)
+- [Comparison with other testing frameworks](#comparison-with-other-testing-frameworks)
+- [Next steps](#next-steps)
+
+## Why testing matters
+
+Testing your code allows you to catch bugs and issues before they reach
+production, saving time and resources. Tests are also useful to help plan out
+the logic of your application, they can serve as a human readable description of
+how your code is meant to be used.
+
+Having thorough tests allows you to refactor your code with confidence, knowing
+you'll be alerted if something breaks or you introduce a bug or regression.
+
+Deno provides [built-in testing capabilities](/runtime/fundamentals/testing/)
+without requiring external frameworks or tools, making it straightforward to
```

Comment:
Nit: To be fair, you still need to pull `@std/testing` for test assertions

Reply:
have updated